app/Providers/LegacyUserProvider.php

```diff
@@ -170,6 +170,10 @@ public function retrieveByCredentials(array $credentials)
         }
 
         $username = $credentials['username'] ?? null;
+        // get the sAMAccountName from the UserPrincipalName
+        if (strpos($username, "@") && $type == "active_directory") {
```

Comment:
What happens if you remove this bit of code?

Reply:
I think it would broke the Ldap or other mechanism of authentication.

Reply:
no, I mean the code inside the if statement.